File: Practice/leetCode/Kth Largest Element in an Array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        priority_queue<int, vector<int>, greater<int>> minHeap;
        for(auto i:nums){
            minHeap.push(i);
            while(minHeap.size() >k){
                minHeap.pop();
            }
        }
        
        return minHeap.top();
    }
};

File: LeetCode/Kth-Largest-Element-in-an-Array.java
1class Solution {
2    public int findKthLargest(int[] nums, int k) {
3        int n=nums.length;
4        Queue<Integer> q = new PriorityQueue<>();
5
6        for(Integer i:nums){
7            q.add(i);
8        }
9
10        while(q.size() != k)    q.poll();
11
12        return q.poll();
13    }
14}